sources/web/datalab/polymer/test/table-details-test.ts

```diff
@@ -0,0 +1,132 @@
+
+  const mockTable: gapi.client.bigquery.Table = {
+    creationTime: '1501541271001',
+    etag: 'testEtag',
+    id: 'pid:did.tid',
+    kind: 'table',
+    labels: [{
+      name: 'label1',
+      value: 'label1val',
+    }],
+    lastModifiedTime: '1501541271001',
+    location: 'testLocation',
+    numBytes: (1024 * 1023).toString(),
+    numLongTermBytes: (1024 * 1024 * 50).toString(),
+    numRows: '1234567890',
+    schema: {
+      fields: [{
+        mode: 'mode1',
+        name: 'field1',
+        type: 'value1',
+      }, {
+        name: 'field2',
+        type: 'value2',
+      }],
+    },
+    selfLink: 'testLink',
+    tableReference: {
+      datasetId: 'did',
+      projectId: 'pid',
+      tableId: 'tid',
+    },
+    type: 'table',
+  };
+
+  /**
+   * Rows must be recreated on each test with the fixture, to avoid state leakage.
+   */
+  beforeEach(() => {
+    GapiManager.getBigqueryTableDetails = (pid, did, tid) => {
+      assert(!!pid && !!did && !!tid,
+          'getTableDetails should be called with project, dataset, and table');
+
+      const request = {
+        result: mockTable,
+      };
+      return Promise.resolve(request) as any;
+    };
+
+    testFixture = fixture('table-details-fixture');
+    testFixture.tableId = '';
+    Polymer.dom.flush();
+  });
+
+  it('displays an empty element if no table is provided', () => {
+    assert(!testFixture.$.placeholder.hidden, 'placeholder should be shown');
+    assert(testFixture.$.container.hidden, 'container should be hidden');
+    assert(testFixture.$.placeholder.querySelector('paper-spinner'),
+        'spinner should be hidden if no table is loading');
+  });
+
+  it('loads the table given its id and gets its details', (done: () => null) => {
+    testFixture.addEventListener('_table-changed', () => {
+      Polymer.dom.flush();
+
+      assert(JSON.stringify((testFixture as any)._table) === JSON.stringify(mockTable),
+          'element should have fetched table info');
+
+      assert(testFixture.$.tableId.innerText === mockTable.tableReference.tableId,
+            'table id should be parsed');
+      assert(testFixture.$.projectId.innerText === mockTable.tableReference.projectId,
+            'project id should be parsed');
+      assert(testFixture.$.datasetId.innerText === mockTable.tableReference.datasetId,
+            'dataset id should be parsed');
+      assert(testFixture.$.tableSize.innerText === '1023.00 KB', 'should see readable table size');
+      assert(testFixture.$.longTermTableSize.innerText === '50.00 MB',
+          'should see readable long term table size');
+      assert(testFixture.$.numRows.innerText === '1,234,567,890',
+          'should see comma-separated number of rows');
+      assert(testFixture.$.creationTime.innerText === '7/31/2017, 3:47:51 PM',
+          'should parse timestamp into readable text');
+      assert(testFixture.$.lastModifiedTime.innerText === '7/31/2017, 3:47:51 PM',
```

Comment:
Would be better if the lastModifiedTime were different from the lastCreationTime.